src/NuGet.Core/NuGet.Packaging/Licenses/NuGetLicenseData.cs

```diff
@@ -13,7 +13,7 @@ public static class NuGetLicenseData
         // Refer to https://github.com/spdx/LicenseListPublisher/issues/102. 
         // When this files gets regenerated, this explanation will be removed. 
         // This is to be used as a marker to confirm that the linked issue with the data source has been fixed.
-        public static string LicenseListVersion = "da7ecca";
+        public const string LicenseListVersion = "da7ecca";
```

Comment:
This value is not a constant. It changes each time the SPDX owners update their data source, and then we run our tool to regenerate this file.

Making it a const will mean that code that reference this value, but then have a different version of `NuGet.Packaging` brought in because of dependencies, will have mismatched `LicenseListVersion` compared to the `LisenseList` below.

Making `LicenseListVersion` a get-only property would be better.

Reply:
AFAIK, we don't need to change `LicenseListVersion` to `readonly` as per my understanding.
- To update `NuGetLicenseData` class file we execute [`$repositoryRoot/scripts/utils/UpdateNuGetLicenseSPDXList.ps1`](https://github.com/NuGet/NuGet.Client/blob/dev/scripts/utils/UpdateNuGetLicenseSPDXList.ps1)
- `$targetFile` parameter in the above script is set to `src\NuGet.Core\NuGet.Packaging\Licenses\NuGetLicenseData.cs`
- The contents of the `$targetFile` are overwritten [here](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestExtensions/GenerateLicenseList/Program.cs#L38). The [File.CreateText(String) Method doc](https://docs.microsoft.com/en-us/dotnet/api/system.io.file.createtext?view=net-5.0) says `If the file already exists, its contents are overwritten`.
- I modified the logic that generates the class file in this PR to always mark the variable as `const` https://github.com/NuGet/NuGet.Client/pull/4028/files#diff-26391a81ca1c15b62f62a482e277541b458d7edd4013b2f12e4fefb82caedb9cR123

Please let me know if I am missing something. I am happy to learn more and change the code accordingly.

Reply:
How the file or value gets updated doesn't matter to customers using our package. It's just our own implementation detail, which we can change.

From the customer point of view:

|If `LicenseListVersion` is|then|
|---|---|
|a writable field|The compiler won't protect them, or any other library they use, from accidentally setting a different value.|
|A const field|The value they compiled against gets "burned in" to their assembly, and if at runtime they actually use a different version of NuGet.Packaging with a different version of the SPDX data, then the version that's burned in to their assembly won't match the version burned in our assembly.|
|A get only property|The version of the SPDX data used in NuGet.Packaging will always be retrieved, even if a different version of NuGet.Packaging is used at runtime, compared to at compile time.|
|A readonly field|Like the read-only property it will always get the right version at runtime, not compile time. But it prevents refactoring of the public API where the implementation might not want to be a field, or might want to be stored somewhere else. A property is just a fancy method, and the method/property getter can be modified during a refactoring to maintain API and ABI compatibility, whereas to maintain API and ABI compatibility with a field means the field can never change. In tight loops, a field will have better perf than a method, but that's not a concern here.|


In short, change `LicenseListVersion` to `public static string LicenseListVersion { get; } = "abc123";` or `public static string LicenseListVersion { get => "abc123" }`

Reply:
Thank you Andy, for the detailed explanation. I changed the field to a `get only property`.